code/modules/cm_aliens/hivebuffs/hivebuff.dm

```diff
@@ -0,0 +1,517 @@
+				if(target_area.flags_area & AREA_NOTUNNEL)
+					failed = TRUE		
+					break			
+		if(!failed)
+			spawn_turf = potential_turf
+			break
+
+	if(!spawn_turf)
+		engage_failure_message = "Unable to find a viable spawn point for the Destroyer"
+		return FALSE
+
+	new /obj/effect/alien/resin/destroyer_cocoon(spawn_turf)
+
+	return TRUE
+
+/datum/hivebuff/defence
+	name = "Boon of Defence"
+	desc = "Increases all xenomorph armour by 5% for 5 minutes"
+	tier = HIVEBUFF_TIER_MINOR
+
+	engage_flavourmessage = "The Queen has imbued us with greater chitin."
+	duration = 5 MINUTES
+	number_of_required_pylons = 1
+	radial_icon = "shield"
+
+/datum/hivebuff/defence/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection += 0.05 * xeno.armor_deflection
+
+/datum/hivebuff/defence/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection = initial(xeno.armor_deflection)
+
+/datum/hivebuff/defence/major
+	name = "Major Boon of Defence"
+	desc = "Increases all xenomorph armour by 10% for 10 minutes"
+	tier = HIVEBUFF_TIER_MAJOR
+
+	engage_flavourmessage = "The Queen has imbued us with even greater chitin."
+	duration = 10 MINUTES
+	cost = 2
+	number_of_required_pylons = 2
+	radial_icon = "shield_m"
+
+/datum/hivebuff/defence/major/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection += 0.1 * xeno.armor_deflection
+
+/datum/hivebuff/defence/major/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.armor_deflection = initial(xeno.armor_deflection)
+
+/datum/hivebuff/attack
+	name = "Boon of Aggression"
+	desc = "Increases all xenomorph damage by 5% for 5 minutes"
+	tier = HIVEBUFF_TIER_MINOR
+
+	engage_flavourmessage = "The Queen has imbued us with slarp claws."
+	duration = 5 MINUTES
+	number_of_required_pylons = 1
+	radial_icon = "slash"
+
+/datum/hivebuff/attack/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.melee_damage_lower += 0.05 * xeno.melee_damage_lower
+	xeno.melee_damage_upper += 0.05 * xeno.melee_damage_upper
+
+/datum/hivebuff/attack/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.melee_damage_lower += initial(xeno.melee_damage_lower)
+	xeno.melee_damage_upper += initial(xeno.melee_damage_upper)
+
+/datum/hivebuff/attack/major
+	name = "Major Boon of Aggression"
+	desc = "Increases all xenomorph damage by 10% for 10 minutes"
+	tier = HIVEBUFF_TIER_MAJOR
+
+	engage_flavourmessage = "The Queen has imbued us with razer sharp claws."
+	duration = 10 MINUTES
+	number_of_required_pylons = 2
+	cost = 2
+	radial_icon = "slash_m"
+
+/datum/hivebuff/attack/major/apply_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.melee_damage_lower += 0.1 * xeno.melee_damage_lower
+	xeno.melee_damage_upper += 0.1 * xeno.melee_damage_upper
+
+/datum/hivebuff/attack/major/remove_buff_effects(mob/living/carbon/xenomorph/xeno)
+	xeno.melee_damage_lower += initial(xeno.melee_damage_lower)
+	xeno.melee_damage_upper += initial(xeno.melee_damage_upper)
```

Comment:
These modifications should be applied to `damage_modifier` as + / - because you will lose the melee_damage_lower 
 modification and melee_damage_upper  once `recalculate_damage` is called.